src/Graphql/Type/SchemaBuilder.php

```diff
@@ -294,16 +294,18 @@ private function convertType(Type $type, bool $input = false, string $mutationNa
      */
     private function getResourceObjectType(string $resourceClass, ResourceMetadata $resourceMetadata, bool $input = false, string $mutationName = null): GraphQLType
     {
-        $shortName = $resourceMetadata->getShortName();
-        if ($input) {
-            $shortName .= 'Input';
+        if (isset($this->graphqlTypes[$resourceClass][$mutationName][$input])) {
+            return $this->graphqlTypes[$resourceClass][$mutationName][$input];
         }
+
+        $shortName = $resourceMetadata->getShortName();
         if (null !== $mutationName) {
-            $shortName .= ucfirst($mutationName).'Mutation';
+            $shortName = $mutationName.ucfirst($shortName);
         }
-
-        if (isset($this->graphqlTypes[$resourceClass][$mutationName][$input])) {
-            return $this->graphqlTypes[$resourceClass][$mutationName][$input];
+        if ($input) {
+            $shortName .= 'Input';
+        } elseif (null !== $mutationName) {
+            $shortName .= 'Payload';
         }
```

Comment:
May this be shorter using `sprintf`?

Reply:
Not sure it will be more clear as `$shortName` is built from conditions

Reply:
Indeed, `sprinf` doesn't help.